parlai/scripts/detect_offensive_language.py

```diff
@@ -59,39 +69,63 @@ def detect(opt, printargs=None, print_parser=None):
         log_every_n_secs = float('inf')
     log_time = TimeLogger()
 
-    # Show some example dialogs:
-    cnt = 0
+    stats = {
+        'bad_words': [],
+        'bad_words_cnt': 0,
+        'string_offensive': 0,
+        'classifier_offensive': 0,
+        'total_offensive': 0,
+        'total': 0,
+    }
+
+    def report(world, stats):
+        report = world.report()
+        log = {
+            'word_offenses': stats['bad_words_cnt'],
+            'classifier_offenses%': 100
+            * (stats['classifier_offensive'] / stats['total']),
+            'string_offenses%': 100 * (stats['string_offensive'] / stats['total']),
+            'total_offenses%': 100 * (stats['total_offensive'] / stats['total']),
+        }
+        text, log = log_time.log(report['exs'], world.num_examples(), log)
```

Comment:
q: what does this log_time.log do? does it make the `log` pretty?

Reply:
hmm.. not sure.. i think this was there already.. i guess it prints it in some format, yes..!